### PR TITLE
Added support for kicad-5.1.x

### PR DESCRIPTION
**Pull request by VedantParanjape**
* Changed if-else handing of args.outMode to args.outMode.find("ki") such that, ki version could be grouped together under one block
* Changed part where polygon data was written to gen according to 5.1.x format
* Added option to arg parser to generate kicad-5.1.x file

Signed-off-by: Vedant Paranjape <22630228+VedantParanjape@users.noreply.github.com>

### CODE REVIEW

**Comment by gregdavill** [comment]
Thanks! Overall this looks good.
But I want to wait until my code has been merged here: https://github.com/sparkfunX/Buzzard/pull/21 before we merge this. 

My feeling is that it could make changing things more difficult if this were applied while that upstream pull-request is still open.

**Comment by VedantParanjape** [comment]
@gregdavill Do you know when this PR will be merged ? I think Buzzard maintainer is busy with something else

**Comment by gregdavill** [comment]
I'll give them another week.

**Comment by VedantParanjape** [comment]
Hi @gregdavill the PR seems to be merged, can this be merged now and can you send a PR to the main repository ?

File: buzzard.py [hunk 1]
```diff
@@ -585,7 +586,12 @@ def drawSVG(svg_attributes, attributes, paths):
         out += "(footprint \"buzzardLabel\"\n" + \
             " (layer \"F.Cu\")\n" + \
             " (attr board_only exclude_from_pos_files exclude_from_bom)\n"
-
+    if args.outMode == "ki-5.1.x":
```
Comment (gregdavill) [inline buzzard.py]: Does this have to be so specific? Can it simply be 'ki5'? From my understanding this should work any version of KiCad 5.x.x

File: buzzard.py [hunk 2]
```diff
@@ -1106,8 +1116,8 @@ def simplify(points, tolerance, highestQuality):
     parser.add_argument('-v', dest='verbose', default=False,
                         help='Verbose mode (helpful for debugging)', action='store_true')
 
-    parser.add_argument('-o', dest='outMode', default='b', choices=['b', 'ls', 'lib', 'ki'],
-                        help='Output Mode (\'b\'=board script, \'ls\'=library script, \'lib\'=library file, \'ki\'=KiCad footprint)')
+    parser.add_argument('-o', dest='outMode', default='b', choices=['b', 'ls', 'lib', 'ki', 'ki-5.1.x'],
+                        help='Output Mode (\'b\'=board script, \'ls\'=library script, \'lib\'=library file, \'ki\'=KiCad nightly(5.99) footprint, \'ki-5.1.x\'=Kicad 5.1.x footprint)')
```
Comment (gregdavill) [inline buzzard.py]: We can probably just call this KiCad v6 and KiCad v5. I know v6 isn't out yet, but the file format in Nightly is what v6 will use.

**Review by gregdavill — CHANGES_REQUESTED** [review]
Just a few small changes, other than that it looks fine to merge.

File: buzzard.py (hunk 1 above)
Comment (VedantParanjape) [reply buzzard.py]: I was not sure, so made it 5.1.x. I'll do the required change

File: buzzard.py (hunk 2 above)
Comment (VedantParanjape) [reply buzzard.py]: Sure

**Comment by VedantParanjape** [comment]
@gregdavill I have made the required changes

File: buzzard.py [hunk 3]
```diff
@@ -1106,8 +1116,8 @@ def simplify(points, tolerance, highestQuality):
     parser.add_argument('-v', dest='verbose', default=False,
                         help='Verbose mode (helpful for debugging)', action='store_true')
 
-    parser.add_argument('-o', dest='outMode', default='b', choices=['b', 'ls', 'lib', 'ki'],
-                        help='Output Mode (\'b\'=board script, \'ls\'=library script, \'lib\'=library file, \'ki\'=KiCad footprint)')
+    parser.add_argument('-o', dest='outMode', default='b', choices=['b', 'ls', 'lib', 'ki', 'ki5'],
+                        help='Output Mode (\'b\'=board script, \'ls\'=library script, \'lib\'=library file, \'ki\'=KiCad v6 footprint, \'ki5\'=Kicad v5 footprint)')
```
Comment (gregdavill) [inline buzzard.py]: Very small nit-pick here.

Just on the capitalization of KiCad, to match the v6 description
```diff
- \'ki5\'=Kicad v5 footprint`
+ \'ki5\'=KiCad v5 footprint`
```

Comment (VedantParanjape) [reply buzzard.py]: @gregdavill Done !

**Review by gregdavill — APPROVED** [review]

**Comment by VedantParanjape** [comment]
Hi @gregdavill When will this change be submitted as PR to upstream repository ? If done, we can change the submodule in deps folder to the upstream buzzard repository

**Pull request merged**